### PR TITLE
Add homework1

### PR DESCRIPTION
**Pull request by HuMMaI**
Added first Java Advanced homework

### CODE REVIEW

File: src/main/java/homework1/HoursException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package homework1;
+
+public class HoursException extends Exception {
```
Comment (VolodymyrTymtsiv) [inline src/main/java/homework1/HoursException.java]: We could extend from `RuntimeException` that allows us to nut add `throws HoursException` to methods.

File: src/test/java/homework1Test/AccountingTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        int actual = accounting.salaryCalculation();
+        int expected = 3400 - accounting.getFine();
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test(expected = HoursException.class)
+    public void itSalaryCalculationWithHoursException() throws HoursException {
+        accounting = new Accounting(20, 121, 120,
+                5, 10);
```
Comment (VolodymyrTymtsiv) [inline src/test/java/homework1Test/AccountingTest.java]: You don't need to reassign `accounting`, it is already resigned in `setup()` method before each test.

File: src/test/java/homework1Test/FactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    @Test(expected = NullPointerException.class)
+    public void itAddSalaryToWorkerWithException() throws HoursException {
+        Accounting accounting = new Accounting(20, 120, 120,
+                5, 10);
+        factory.addSalaryToWorker(null, accounting);
+    }
+
+    @Test
+    public void itOutputAllWorkers() throws HoursException {
+        Accounting accounting = new Accounting(20, 120, 120,
```
Comment (VolodymyrTymtsiv) [inline src/test/java/homework1Test/FactoryTest.java]: Here as well.

File: src/main/java/homework1/HoursException.java (hunk 1 above)
Comment (HuMMaI) [reply src/main/java/homework1/HoursException.java]: Done

File: src/test/java/homework1Test/FactoryTest.java (hunk 3 above)
Comment (HuMMaI) [reply src/test/java/homework1Test/FactoryTest.java]: Done

File: src/test/java/homework1Test/AccountingTest.java (hunk 2 above)
Comment (HuMMaI) [reply src/test/java/homework1Test/AccountingTest.java]: Done

**Pull request merged**